tests/test_schema_errors/test_objects.py

```diff
@@ -0,0 +1,52 @@
+import pytest
+
+from openapi_tester import DocumentationError, SchemaTester
+
+example_schema_object = {"type": "object", "properties": {"value": {"type": "integer"}}}
+example_object = {"value": 1}
+
+tester = SchemaTester()
+
+
+def test_nothing_wrong():
+    """ This should always pass """
+    tester.test_schema_section(example_schema_object, example_object, "")
+
+
+def test_response_is_missing_keys():
+    """ Missing keys in a response object should raise an error """
```

Comment:
nope, this test is too crude. You should read the logic and test in a more subtle and elaborate way.

Reply:
How is it too crude? This seems completely fundamental to me, if we're attempting any sort of validation. If an API is documented to return something, and it isn't returned,  we should raise an exception.

Are you agreeing with the above, but suggesting this isn't a good way to test this, because of how we've structured the new validation?

Reply:
yes, i agree we should test this. But the test is crude vis-a-vis the validation logic. 

We are using the required keyword to decide what is required and what isn't. Only if a key is specified in the required list, or if no required list is supplied (we then assume everything is required) do we regard it as missing.

Reply:
Nice, change it to this

```python
def test_response_is_missing_keys():
    # If a required key is missing, we should raise an error
    required_key = {"type": "object", "properties": {"value": {"type": "integer"}}, 'required': ['value']}
    with pytest.raises(DocumentationError, match="The following properties are missing from the tested data: value"):
        tester.test_schema_section(required_key, {})

    # If not required, it should pass
    optional_key = {"type": "object", "properties": {"value": {"type": "integer"}}}
    tester.test_schema_section(optional_key, {})
```